src/extra/JsonSchemaRenderer/widgets/ui-options.ts

```diff
@@ -0,0 +1,142 @@
+import { JSONSchema } from '../types';
+
+export type UiOptions = {
+	[key: string]: JSONSchema;
+};
+
+// styled-system props can be passed as arrays, with each item
+// representing the value at a particular breakpoint.
+const responsive = (schema: JSONSchema): JSONSchema => {
+	return {
+		anyOf: [
+			{
+				type: 'array',
+				items: schema,
+			},
+			{
+				...schema,
+			},
+		],
+	};
+};
+
+// Convenience object for common UI option schemas
+export const UiOption: UiOptions = {
+	string: {
+		type: 'string',
+	},
+	boolean: {
+		type: 'boolean',
+	},
+	number: {
+		type: 'number',
+	},
+	integer: {
+		type: 'integer',
+	},
+	space: responsive({
+		type: ['number', 'string'],
+	}),
+};
+
+export const SpaceUiOptions: UiOptions = {};
+
+[
+	'p',
+	'pt',
+	'pb',
+	'pl',
+	'pr',
+	'px',
+	'py',
+	'm',
+	'mt',
+	'mb',
+	'ml',
+	'mr',
+	'mx',
+	'my',
+].reduce((options, option) => {
+	options[option] = UiOption.space;
+	return options;
+}, SpaceUiOptions);
+
+export const FlexUiOptions: UiOptions = {
```

Comment:
This is not great, but I can't think of a better approach :)

Reply:
Ha - totally agree! Got me thinking about a Typescript <--> JSONSchema converter!